File: app/src/test/java/tw/ctl/interest/calculation/CalculationPresenterTest.java
package tw.ctl.interest.calculation;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import tw.ctl.interest.Entity;

import static org.mockito.Mockito.verify;

/**
 * Created by jacklai on 2017/3/1.
 */

public class CalculationPresenterTest {

    @Mock
    private CalculationView view;

    private CalculationPresenter presenter;

    @Before
    public void setUp() throws Exception {
        MockitoAnnotations.initMocks(this);
        presenter = new CalculationPresenter();
        presenter.attachView(view);
    }

    @After
    public void tearDown() throws Exception {
        presenter.detachView();
    }

    @Test
    public void calculate() throws Exception {
        Entity entity = new Entity();
        entity.principal = "1000";
        entity.interest  = "10";
        entity.period    = "10";
        entity.invest    = "1000";

        presenter.calculate(entity);

        entity.simpleResult = "2,000.00";
        entity.compoundResult = "2,593.74";
        entity.investResult = "18,531.17";

        verify(view).onResult(entity);
    }

}

File: app/src/test/java/tw/ctl/interest/calculation/CalculationPresenterTest.kt
package tw.ctl.interest.calculation

import org.junit.After
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify
import tw.ctl.interest.model.Record

class CalculationPresenterTest {

    private val view = mock(CalculationView::class.java)
    private var presenter: CalculationPresenter? = null

    @Before
    @Throws(Exception::class)
    fun setUp() {
        presenter = CalculationPresenter()
        presenter?.attachView(view)
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        presenter?.detachView()
    }

    @Test
    @Throws(Exception::class)
    fun calculateWithInvest() {
        val record = Record("1000", "10", "10", "1000")
        presenter?.calculate(record)

        record.simpleResult = "2,000.00"
        record.compoundResult = "2,593.74"
        record.investResult = "18,531.17"

        verify<CalculationView>(view).onResult(record)
    }

    @Test
    @Throws(Exception::class)
    fun calculateWithoutInvest() {
        val record = Record("1000", "10", "10", "")
        presenter?.calculate(record)

        record.simpleResult = "2,000.00"
        record.compoundResult = "2,593.74"
        record.investResult = "---"

        verify<CalculationView>(view).onResult(record)
    }

}